openfermioncirq/gates/fermionic_simulation.py

```diff
@@ -145,10 +208,21 @@ def fswap(self, i: int):
         """Update the weights of the gate to effect conjugation by an FSWAP on
         the i-th and (i+1)th qubits."""
 
-    @property
-    def num_weights(self) -> int:
+    @classmethod
+    @abc.abstractmethod
+    def from_interaction_operator(
```

Comment:
This method should not be defined on ParityPreservingFermionicGate. For instance, a subclass might use only higher-order terms that cannot be present in an InteractionOperator.

Reply:
I introduced a new class `InteractionOperatorFermionicGate` for this and the newly named `interaction_operator_generator`. Very open to a better name.

It's nice to have a class for two reasons: 
1) Type checking (not just mypy, but at runtime in methods like `fermionic_simulation_gates_from_interaction_operator`).
2) Enforcing signature consistency.